### PR TITLE
Document qarray

### PR DESCRIPTION
**Pull request by pierreguilmin**
First version of the documentation.

### CODE REVIEW

File: dynamiqs/qarrays/qarray.py [hunk 1]
```diff
@@ -66,10 +66,61 @@ def _dims_from_qutip(dims: list[list[int]]) -> tuple[int, ...]:
...
+        layout _(Layout)_: Data layout, either `dq.dense` or `dq.dia`.
+        dims _(tuple of ints)_: Hilbert space dimension of each subsystem.
+        mT _(QArray)_: Returns the qarray transposed over its last two dimensions.
+
+    Note: Arithmetic operation support
+        Qarrays support elementary operations, such as element-wise
+        addition/subtraction, element-wise multiplication and matrix multiplications
+        with other qarray-like objects.
+
+    Note-: Shortcuts methods to use quantum utilities
```
Comment (gautierronan) [inline dynamiqs/qarrays/qarray.py:96]: I think this note and its table are quite important (equally to the rest of the docstring IMO). I would propose not having it as a note. This is what PyTorch does for instance, [here](https://pytorch.org/docs/stable/tensors.html#torch.Tensor.mH).

Actually, I would even go further and provide a table for the _all_ methods of a `dq.QArray`, such that:

- for methods that are duplicates of quantum utilities (e.g. `.dag()`), just have a description saying "See `dq.this`" or "Alias of `dq.this`" (like pytorch does)
- for methods that are new to QArrays (e.g. `.conj()` or `reshape(*shape)`), have a short one-liner description + link to the full docstring just below in the page.

File: dynamiqs/qarrays/qarray.py [hunk 2]
```diff
@@ -66,10 +66,61 @@ def _dims_from_qutip(dims: list[list[int]]) -> tuple[int, ...]:
...
+        Many functions of the library can be called directly on a qarray rather than
+        through the functional API. For example, you can use `x.dag()` instead of
+        `dq.isdag(x)`. Here is the complete list of these shortcuts:
```
Comment (gautierronan) [inline dynamiqs/qarrays/qarray.py:99]:
```suggestion
        `dq.dag(x)`. Here is the complete list of these shortcuts:
```
typo

**Review by gautierronan — CHANGES_REQUESTED** [review]

**Comment by pierreguilmin** [comment]
@gautierronan if you agree with the overall content I propose to merge this and iterate on top later for styling / minor modifications.

**Review by gautierronan — APPROVED** [review]
LGTM!

**Pull request merged**